server/game/cards/plots/07/thewhiteshadows.js

```diff
@@ -0,0 +1,28 @@
+const PlotCard = require('../../../plotcard.js');
+
+class TheWhiteShadows extends PlotCard {
+    setupCardAbilities() {
+        this.reaction({
+            when: {
+                onCharacterKilled: event => this.controller !== event.card.controller
+            },
+            handler: context => {
+                this.controller.putIntoPlay(context.event.card);
+                this.game.addMessage('{0} uses {1} to put {2} into play under their control, blank', 
+                                      this.controller, this, context.event.card);
+
+                this.untilEndOfPhase(ability => ({
+                    match: context.event.card,
+                    effect: [
+                        ability.effects.blank,
+                        ability.effects.moveToDeadPileIfStillInPlay()
```

Comment:
This should probably be moved to a separate `atEndOfPhase` effect to match the card's text (not sure if there are scenarios where it could make a difference since the card is also blank, but there may be)